src/ipc/dbus/DBusObjectRegistry.cpp

```diff
@@ -111,32 +145,31 @@ void DBusObjectRegistry::MasterImpl::init()
 
 bool DBusObjectRegistry::MasterImpl::registerObject(const QString &objectPath, const QString &serviceName)
 {
-    auto objects = m_objects.value();
-    if (!objects.contains(objectPath)) {
-        qCDebug(LogIpc) << "Object registered at path" << objectPath << "service name:" << serviceName;
-        objects[objectPath] = serviceName;
-        m_objects = objects;
+    if (!m_objectMap.contains(objectPath) && !objectPath.isEmpty() && !serviceName.isEmpty()) {
+        m_objectMap.insert(objectPath, serviceName);
+        qCDebug(LogIpc) << "Object registered at path" << objectPath
+                        << "service name:" << serviceName;
+        emit objectRegistered(objectPath, serviceName);
         return true;
+    } else if (objectPath.isEmpty() || serviceName.isEmpty()) {
```

Comment:
this condition already appears above. Avoid duplication by checking if we are in that case first.

Reply:
this is not the same condition. See De Morgan's laws. I will change it a bit.

Reply:
Changed.